lnwallet/commitment.go

```diff
@@ -0,0 +1,450 @@
+	// need the objective local/remote keys for this particular commitment
+	// as well. For any non-dust HTLCs that are manifested on the commitment
+	// transaction, we'll also record its CLTV which is required to sort the
+	// commitment transaction below. The slice is initially sized to the
+	// number of existing outputs, since any outputs already added are
+	// commitment outputs and should correspond to zero values for the
+	// purposes of sorting.
+	cltvs := make([]uint32, len(commitTx.TxOut))
+	for _, htlc := range filteredHTLCView.ourUpdates {
+		if htlcIsDust(false, isOurs, feePerKw,
+			htlc.Amount.ToSatoshis(), dustLimit) {
+			continue
+		}
+
+		err := addHTLC(commitTx, isOurs, false, htlc, keyRing)
+		if err != nil {
+			return nil, err
+		}
+		cltvs = append(cltvs, htlc.Timeout)
+	}
+	for _, htlc := range filteredHTLCView.theirUpdates {
+		if htlcIsDust(true, isOurs, feePerKw,
+			htlc.Amount.ToSatoshis(), dustLimit) {
+			continue
+		}
+
+		err := addHTLC(commitTx, isOurs, true, htlc, keyRing)
+		if err != nil {
+			return nil, err
+		}
+		cltvs = append(cltvs, htlc.Timeout)
+	}
+
+	// Set the state hint of the commitment transaction to facilitate
+	// quickly recovering the necessary penalty state in the case of an
+	// uncooperative broadcast.
+	err = SetStateNumHint(commitTx, height, cb.obfuscator)
+	if err != nil {
+		return nil, err
+	}
+
+	// Sort the transactions according to the agreed upon canonical
+	// ordering. This lets us skip sending the entire transaction over,
+	// instead we'll just send signatures.
+	InPlaceCommitSort(commitTx, cltvs)
+
+	// Next, we'll ensure that we don't accidentally create a commitment
+	// transaction which would be invalid by consensus.
+	uTx := btcutil.NewTx(commitTx)
+	if err := blockchain.CheckTransactionSanity(uTx); err != nil {
+		return nil, err
+	}
+
+	// Finally, we'll assert that were not attempting to draw more out of
+	// the channel that was originally placed within it.
+	var totalOut btcutil.Amount
+	for _, txOut := range commitTx.TxOut {
+		totalOut += btcutil.Amount(txOut.Value)
+	}
+	if totalOut > cb.chanState.Capacity {
+		fmt.Errorf("height=%v, for ChannelPoint(%v) attempts "+
+			"to consume %v while channel capacity is %v",
+			height, cb.chanState.FundingOutpoint,
+			totalOut, cb.chanState.Capacity)
+	}
+
+	return &commitmentView{
+		txn:          commitTx,
+		fee:          commitFee,
+		ourBalance:   ourBalance,
+		theirBalance: theirBalance,
+	}, nil
+}
+
+// CreateCommitTx creates a commitment transaction, spending from specified
+// funding output. The commitment transaction contains two outputs: one local
+// output paying to the "owner" of the commitment transaction which can be
+// spent after a relative block delay or revocation event, and a remote output
+// paying the counterparty within the channel, which can be spent immediately
+// or after a delay depending on the commitment type..
+func CreateCommitTx(chanType channeldb.ChannelType,
```

Comment:
Shouldn't this also be a method on `CommitmentBuilder`?

Reply:
Not sure what we gain by doing that, as there are less dependencies without (it only need the `ChannelType` instead of the whole `OpenChannel`)

Reply:
My idea is that that would prevent pass in wrong parameters here. Or is this called from places where we don't have a `CommitmentBuilder` yet?